optaplanner-examples/src/test/java/org/optaplanner/examples/common/app/SolverSmokeTest.java

```diff
@@ -0,0 +1,241 @@
+                streamBuilder.add(createSpeedTest(testData.constraintStreamImplType, testData.unsolvedDataFile,
+                        FULL_ASSERT,
+                        testData.bestScoreLimitForFullAssert, moveThreadCount));
+            }
+            return streamBuilder.build();
+        }));
+    }
+
+    private DynamicTest createSpeedTest(ConstraintStreamImplType constraintStreamImplType, String unsolvedDataFile,
+            EnvironmentMode environmentMode, Score_ bestScoreLimit, String moveThreadCount) {
+        String testName = constraintStreamImplType + ", " +
+                unsolvedDataFile.replaceFirst(".*/", "")
+                + ", "
+                + environmentMode
+                + ", threads: " + moveThreadCount;
+        return dynamicTest(testName,
+                () -> runSpeedTest(constraintStreamImplType,
+                        new File(unsolvedDataFile),
+                        bestScoreLimit,
+                        environmentMode,
+                        moveThreadCount));
+    }
+
+    @TestFactory
+    @Execution(ExecutionMode.CONCURRENT)
+    @Timeout(600)
+    Stream<DynamicTest> runConstraintStreamsMutualCorrectnessTest() {
+        return testData()
+                .map(testData -> new File(testData.unsolvedDataFile))
+                .distinct()
+                .map(this::createConstraintStreamsMutualCorrectnessTest);
+    }
+
+    private DynamicTest createConstraintStreamsMutualCorrectnessTest(File unsolvedDataFile) {
+        String testName = "DROOLS v. BAVET, " +
+                unsolvedDataFile.toString().replaceFirst(".*/", "")
+                + ", "
+                + FULL_ASSERT;
+        return dynamicTest(testName,
+                () -> runConstraintStreamsMutualCorrectnessTest(unsolvedDataFile));
+    }
+
+    @BeforeEach
+    public void setUp() {
+        CommonApp<Solution_> commonApp = createCommonApp();
+        solutionFileIO = commonApp.createSolutionFileIO();
+        solverConfigResource = commonApp.getSolverConfigResource();
+    }
+
+    protected abstract CommonApp<Solution_> createCommonApp();
+
+    protected abstract Stream<TestData<Score_>> testData();
+
+    private void runSpeedTest(ConstraintStreamImplType constraintStreamImplType, File unsolvedDataFile, Score_ bestScoreLimit,
+            EnvironmentMode environmentMode, String moveThreadCount) {
+        SolverFactory<Solution_> solverFactory =
+                buildSolverFactory(constraintStreamImplType, bestScoreLimit, environmentMode, moveThreadCount);
+        Solution_ problem = solutionFileIO.read(unsolvedDataFile);
+        logger.info("Opened: {}", unsolvedDataFile);
+        Solver<Solution_> solver = solverFactory.buildSolver();
+        Solution_ bestSolution = solver.solve(problem);
+        assertScoreAndConstraintMatches(solverFactory, bestSolution, bestScoreLimit);
+    }
+
+    private SolverFactory<Solution_> buildSolverFactory(ConstraintStreamImplType constraintStreamImplType,
```

Comment:
(does not block merging) Building the `SolverFactory` is a relatively expensive operation, maybe we could cache it ` @BeforeAll` to avoid rebuilding it for every test data combination? This applies to `buildMutualCorrectnessSolverFactory` as well.